File: go-client-sdk/pkg/models/shared/codatpublicapimodelsdataaccountresponse.go
package shared

import (
	"time"
)

type CodatPublicAPIModelsDataAccountResponse struct {
	Currency               *string                                              `json:"currency,omitempty"`
	CurrentBalance         *float64                                             `json:"currentBalance,omitempty"`
	Description            *string                                              `json:"description,omitempty"`
	FullyQualifiedCategory *string                                              `json:"fullyQualifiedCategory,omitempty"`
	FullyQualifiedName     *string                                              `json:"fullyQualifiedName,omitempty"`
	ID                     *string                                              `json:"id,omitempty"`
	IsBankAccount          *bool                                                `json:"isBankAccount,omitempty"`
	ModifiedDate           *time.Time                                           `json:"modifiedDate,omitempty"`
	Name                   *string                                              `json:"name,omitempty"`
	NominalCode            *string                                              `json:"nominalCode,omitempty"`
	SourceModifiedDate     *time.Time                                           `json:"sourceModifiedDate,omitempty"`
	Status                 *CodatPublicAPIModelsDataAccountStatusResponseEnum   `json:"status,omitempty"`
	Type                   *CodatPublicAPIModelsDataAccountTypeResponseEnum     `json:"type,omitempty"`
	ValidDatatypeLinks     []CodatPublicAPIModelsDataValidDatatypeLinksResponse `json:"validDatatypeLinks,omitempty"`
}

File: typescript-client-sdk/src/sdk/models/shared/codatpublicapimodelsdataaccountresponse.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatPublicApiModelsDataAccountStatusResponseEnum } from "./codatpublicapimodelsdataaccountstatusresponseenum";
import { CodatPublicApiModelsDataAccountTypeResponseEnum } from "./codatpublicapimodelsdataaccounttyperesponseenum";
import { CodatPublicApiModelsDataValidDatatypeLinksResponse } from "./codatpublicapimodelsdatavaliddatatypelinksresponse";
import { Expose, Transform, Type } from "class-transformer";


export class CodatPublicApiModelsDataAccountResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "currency" })
  currency?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "currentBalance" })
  currentBalance?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "description" })
  description?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "fullyQualifiedCategory" })
  fullyQualifiedCategory?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "fullyQualifiedName" })
  fullyQualifiedName?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "id" })
  id?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "isBankAccount" })
  isBankAccount?: boolean;

  @SpeakeasyMetadata()
  @Expose({ name: "modifiedDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  modifiedDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "name" })
  name?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "nominalCode" })
  nominalCode?: string;

  @SpeakeasyMetadata()
  @Expose({ name: "sourceModifiedDate" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  sourceModifiedDate?: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "status" })
  status?: CodatPublicApiModelsDataAccountStatusResponseEnum;

  @SpeakeasyMetadata()
  @Expose({ name: "type" })
  type?: CodatPublicApiModelsDataAccountTypeResponseEnum;

  @SpeakeasyMetadata({ elemType: CodatPublicApiModelsDataValidDatatypeLinksResponse })
  @Expose({ name: "validDatatypeLinks" })
  @Type(() => CodatPublicApiModelsDataValidDatatypeLinksResponse)
  validDatatypeLinks?: CodatPublicApiModelsDataValidDatatypeLinksResponse[];
}